chainerx_build_helper.py

```diff
@@ -40,6 +41,15 @@ def run(self):
         if cmake_version < '3.1.0':
             raise RuntimeError('CMake >= 3.1.0 is required to build ChainerX')
 
+        self.use_ninja = False
+        prefer_ninja = os.getenv('CHAINERX_CMAKE_GENERATOR', '').lower() == 'ninja'
+        if prefer_ninja:
+            try:
+                out = subprocess.check_output(['ninja', '--version'])
+                self.use_ninja = True
+            except OSError:
+                warnings.warn('Build with ninja is preferred, but ninja cannot be found.')
```

Comment:
Do we have to fall back to other generator in this case?
I think we can just leave it an error (maybe raised later).
The point is, if the user explicitly set `CHAINERX_CMAKE_GENERATOR`, he wants to use it. And it's not this script's responsibility to detect the error.

Reply:
Do you mean we could just append the value `CHAINERX_CMAKE_GENERATOR` (if it's defined) to `-G<what's given>`? That might be cleaner especially in terms of responsibility.

Reply:
Yes, something like that.

Or perhaps we can just introduce `CHAINERX_CMAKE_EXTRA_FLAGS` which user would set `-GNinja`. This way we can provide more flexibility.
But maybe it's not possible because we need this logic?
```py
            if not self.use_ninja:
                if sys.maxsize > 2**32:
                    cmake_args += ['-A', 'x64']
                build_args += ['--', '/m']
```

Reply:
Hm, yes we do have various logic here so it might be this script's responsibility to act depending on ChainerX's environment variables after all. If we allow more flexibility in what user's can specify though these environment variables, we'll probably have to write more logic to parse it etc.

Reply:
OK, I don't care about flexibility as long as the check is not done here.

Reply:
ninja and the platform default generator will be enough. I never use other generators, even ninja usage is rare. So lets don't bother with them for them for now. As for warning/error, I prefer warning, because for large amount of user, building failure is frustrating and headless.

Reply:
I don't want to check here because:
(1) We can't be really sure if this logic is correct to check availability of `ninja`. CMake's side could possibly have some mechanism to change the executable name from default, like by another environment variable, in which case we can't follow the change.
(2) Combined with (1), there seems to be a security concern. We can't be sure if the command `ninja` is really what the user want to execute. It could be an unrelated command.

Reply:
@niboshi you are right, cmake will do it and will generate good error message.
```
CMake Error: CMake was unable to find a build program corresponding to "Ninja".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
```